dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -95,30 +96,36 @@ private void checkUserExists(User user) {
+                                 @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") Integer limit,
+                                 @QueryParam("offset") @DefaultValue("0") Integer offset) {
         User userWithSession = this.userDAO.findById(user.getId());
-        return getEventsForUser(userWithSession, eventSearchType, limit, offset);
+        return getEventsForUser(userWithSession, userWithSession, eventSearchType, limit, offset);
     }
 
     @GET
     @Timed
     @UnitOfWork(readOnly = true)
     @Path("/{userId}")
-    @Operation(description = "No authentication.", summary = "Get events based on filter and user id.")
+    @Operation(description = "Optional authentication.", summary = "Get events based on filter and user id.")
     @ApiOperation(value = "List recent events for a user.", notes = "No authentication.", response = Event.class, responseContainer = "List")
     @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "A list of events", content = @Content(mediaType = MediaType.APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = Event.class))))
     @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "", description = "User not found")
-    public List<Event> getUserEvents(@ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId,
-        @NotNull @QueryParam("eventSearchType") EventSearchType eventSearchType,
-        @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") Integer limit,
-        @QueryParam("offset") @DefaultValue("0") Integer offset) {
+    public List<Event> getUserEvents(@Parameter(hidden = true) @ApiParam(hidden = true)  @Auth Optional<User> loggedInUser,
+                                     @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId,
+                                     @NotNull @QueryParam("eventSearchType") EventSearchType eventSearchType,
+                                     @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") Integer limit,
+                                     @QueryParam("offset") @DefaultValue("0") Integer offset) {
         User user = this.userDAO.findById(userId);
         checkUserExists(user);
-        return getEventsForUser(user, eventSearchType, limit, offset);
+        if (loggedInUser.isPresent()) {
+            return getEventsForUser(user, loggedInUser.get(), eventSearchType, limit, offset);
```

Comment:
`Optional` has a method `orElse` that combines the semantics of the `isPresent` and `get`.  Since this function call and the following call are the same except for the Optional-based argument, you can use `orElse` to reduce them to a single function call:
```
return getEventsForUser(user, loggedInUser.orElse(null), eventSearchType, limit, offset);
```